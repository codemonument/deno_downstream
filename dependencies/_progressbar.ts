// export * from "https://deno.land/x/progress@v1.2.4/mod.ts";
// export { default as ProgressBar } from "https://deno.land/x/progress@v1.2.4/mod.ts";

export * from "./progress-fork/mod.ts";
export { default as ProgressBar } from "./progress-fork/mod.ts";
