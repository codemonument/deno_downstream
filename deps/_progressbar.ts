// update to >1.3.0 when pr is merged
export * from "https://deno.land/x/progress@v1.3.7/mod.ts";
export { default as ProgressBar } from "https://deno.land/x/progress@v1.3.7/mod.ts";
