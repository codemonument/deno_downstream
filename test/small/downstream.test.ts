import { assert, assertRejects, assertSnapshot } from "@deps/_testing.std.ts";
import { downstream } from "@mod";
import { File100MB, File50MB, File50MB404 } from "@testutils/testfiles.ts";
import { drainStream } from "@testutils/drainStream.ts";
import { ProgressBar } from "@deps/_progressbar.ts";

/**
 * tc = (Deno) test context
 */
Deno.test(`'downstream' function`, async (tc) => {
  await tc.step(`Rejects when HTTP Status is !== 2XX`, async () => {
    await assertRejects(async () => await downstream(File50MB404));
  });

  await tc.step(
    `Simple streaming download and ignoring the progressStream`,
    async () => {
      const { fileStream, contentLength } = await downstream(
        "http://ipv4.download.thinkbroadband.com/50MB.zip",
      );

      assert(typeof contentLength === "number");
      assert(contentLength > 0);

      // needs to be done to consume the fileStream,
      // otherwise deno test will crash
      await drainStream(fileStream);
    },
  );

  await tc.step(`Returns content size correctly`, async () => {
    const { fileStream, contentLength } = await downstream(
      File50MB,
    );

    const kb = contentLength / 1024;
    const mb = kb / 1024;
    assert(mb === 50); // the 50MB testfile consists of 50MB

    // needs to be done to not crash deno test
    await drainStream(fileStream);
  });

  await tc.step(`Reports Progress correctly`, async () => {
    const progressBar = new ProgressBar({ title: "downloading: ", total: 100 });

    const { progressStream, fileStream } = await downstream(File100MB);
    const progressEvents: string[] = [];

    for await (const progress of progressStream) {
      progressBar.render(Number.parseFloat(progress));
      progressEvents.push(progress);
    }

    await assertSnapshot(tc, progressEvents.length);
    // needs to be done to not crash deno test
    await drainStream(fileStream);
  });
});
