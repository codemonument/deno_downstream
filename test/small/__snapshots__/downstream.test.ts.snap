export const snapshot = {};

snapshot[`'downstream' function 1`] = `6401`;
